### PR TITLE
[addons] move binary addon interface functions to own files / cleanups

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
This includes a change which brings the binary addon interface functions (addon to Kodi) into own files. This is intended to clean up the interface.

In addition, the primary task of the CAddonDll is to load the addon for Kodi and make it available, but not to provide its interface itself.

The interface folder therefore refers purely to the API between addon and Kodi, therefore all API functions should also be in it if possible (partly except for Kodi to addon instances).

Furthermore, it contains some code cleanups, so please review the commits individually for review.

The commits from https://github.com/xbmc/xbmc/pull/17292 are temporarily there (was to avoid conflicts) which disappear when it is in there.

## Motivation and Context
To make the addon interface cleaner

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by phunkyfish — APPROVED** [review]
Nice cleanup.

File: xbmc/addons/interfaces/AddonBase.cpp [hunk 1]
```diff
@@ -192,39 +192,39 @@ void Interface_Base::addon_log_msg(void* kodiBase, const int addonLogLevel, cons
...
+  CLog::Log(logLevel, "AddOnLog: {}: {}", addon->Name(), strMessage);
 }
 
 bool Interface_Base::get_setting_bool(void* kodiBase, const char* id, bool* value)
 {
   CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
   if (addon == nullptr || id == nullptr || value == nullptr)
   {
-    CLog::Log(LOGERROR, "Interface_Base::%s - invalid data (addon='%p', id='%p', value='%p')",
+    CLog::Log(LOGERROR, "Interface_Base::{} - invalid data (addon='%p', id='%p', value='%p')",
```
Comment (garbear) [inline xbmc/addons/interfaces/AddonBase.cpp]: Why does `%p` stay old style?

**Review by garbear — APPROVED** [review]

File: xbmc/addons/interfaces/AddonBase.cpp (hunk 1 above)
Comment (AlwinEsch) [reply xbmc/addons/interfaces/AddonBase.cpp]: Was not 100% sure it makes in this case sense to see better it goes about a pointer, seen also some parts now where not use the `static_cast<void*>` for them an want to change.

Comment (AlwinEsch) [reply xbmc/addons/interfaces/AddonBase.cpp]: Seen during test it is not only a can be with `{}`, it is a must be then.
This was given on test:
```
ERROR: Interface_Base::get_setting_bool - invalid data (addon='%p', id='%p', value='%p')
```
Only the `get_setting_bool` was correct the rest has written `%p` to log.

**Pull request merged**